scripts/olm-create-bundle.sh

```diff
@@ -0,0 +1,122 @@
+  AWS_SERVICE_DOCKER_IMG:                   Specify the docker image for the AWS service.
+                                            This should include the registry, namespace,
+                                            image name, and tag. Takes precedence over
+                                            SERVICE_CONTROLLER_CONTAINER_REPOSITORY
+  SERVICE_CONTROLLER_SOURCE_PATH:           Path to the service controller source code
+                                            repository.
+                                            Default: ../{SERVICE}-controller
+  SERVICE_CONTROLLER_CONTAINER_REPOSITORY   The container repository where the controller exists.
+                                            This should include the registry, namespace, and
+                                            image name.
+                                            Default: $DEFAULT_CONTROLLER_CONTAINER_REPOSITORY
+  BUNDLE_GENERATE_EXTRA_ARGS                Extra arguments to pass into the command
+                                            'operator-sdk generate bundle'.
+"
+
+if [ $# -ne 2 ]; then
+    echo "ERROR: $(basename "$0") accepts two parameters, the SERVICE and VERSION" 1>&2
+    echo "$USAGE"
+    exit 1
+fi
+
+SERVICE=$(echo "$1" | tr '[:upper:]' '[:lower:]')
+VERSION=$2
+BUNDLE_VERSION=${BUNDLE_VERSION:-$VERSION}
+
+
+
+DEFAULT_SERVICE_CONTROLLER_SOURCE_PATH="$ROOT_DIR/../$SERVICE-controller"
+SERVICE_CONTROLLER_SOURCE_PATH=${SERVICE_CONTROLLER_SOURCE_PATH:-$DEFAULT_SERVICE_CONTROLLER_SOURCE_PATH}
+
+BUNDLE_OUTPUT_PATH=${BUNDLE_OUTPUT_PATH:-$SERVICE_CONTROLLER_SOURCE_PATH/olm}
+
+if [[ ! -d $SERVICE_CONTROLLER_SOURCE_PATH ]]; then
+    echo "Error evaluating SERVICE_CONTROLLER_SOURCE_PATH environment variable:" 1>&2
+    echo "$SERVICE_CONTROLLER_SOURCE_PATH is not a directory." 1>&2
+    echo "${USAGE}"
+    exit 1
+fi
+
+# Set controller image. 
+if [ -n "$AWS_SERVICE_DOCKER_IMG" ] && [ -n "$SERVICE_CONTROLLER_CONTAINER_REPOSITORY"] ; then
+  # It's possible to set the repository (i.e. everything except the tag) as well as the
+  # entire path including the tag using AWS_SERVIC_DOCKER_IMG. If the latter is set, it
+  # will take precedence, so inform the user that this will happen in case the usage of
+  # each configurable is unclear before runtime. 
+  echo "both AWS_SERVICE_DOCKER_IMG and SERVICE_CONTROLLER_CONTAINER REPOSITORY are set."
+  echo "AWS_SERVICE_DOCKER_IMG is expected to be more complete and will take precedence."
+  echo "Ignoring SERVICE_CONTROLLER_CONTAINER_REPOSITORY."
+fi
+
+SERVICE_CONTROLLER_CONTAINER_REPOSITORY=${SERVICE_CONTROLLER_CONTAINER_REPOSITORY:-$DEFAULT_SERVICE_CONTROLLER_CONTAINER_REPOSITORY}
+DEFAULT_AWS_SERVICE_DOCKER_IMAGE="$SERVICE_CONTROLLER_CONTAINER_REPOSITORY:$SERVICE-$VERSION"
+AWS_SERVICE_DOCKER_IMG=${AWS_SERVICE_DOCKER_IMG:-$DEFAULT_AWS_SERVICE_DOCKER_IMAGE}
+
+pushd $SERVICE_CONTROLLER_SOURCE_PATH/config/controller 1>/dev/null
+kustomize edit set image controller="$AWS_SERVICE_DOCKER_IMG"
```

Comment:
Ugh this is going to modify the file directly, won't it? ;(

Reply:
Yep =( 

It's similar to what's happening here: https://github.com/aws-controllers-k8s/community/blob/main/scripts/kind-build-test.sh#L213 though that seems to use a temp directory. Maybe I try to go that route instead?

Reply:
This will probably be running in an automated CD system, anyway. So hopefully shouldn't matter. Otherwise this change will appear in a `git status`

Reply:
Sure thing. Happy to tweak this as needed!

Reply:
Yeah, agreed with @RedbackThomson that we should strive to have scripts leave no side-effects if possible. @komish would you mind copying the temp config directory approach that the kind-build-test.sh script uses in order to avoid changing file contents directly?

Reply:
This is done. I generate a `TMP_DIR` and cleanup on script cancelation/end just like the `kind-build-test.sh`. I also added the PRESERVE option, and a user-provided TMP_DIR option.

As an aside, the `ack-generate olm` does generate things in `config/` such as the `manifests/` directory, along with other patches. I'm not sure if these are also considered side effects that we don't want in the base `config/` directory.

We could potentially look at storing these generated things elsewhere and then merging the relevant directories in the `TMP_DIR` when it comes time to generate the bundle, but it would change the logic a bit. 

Not sure if it matters, but figured I'd mention. Let me know.

Reply:
> This is done. I generate a `TMP_DIR` and cleanup on script cancelation/end just like the `kind-build-test.sh`. I also added the PRESERVE option, and a user-provided TMP_DIR option.
> 
> As an aside, the `ack-generate olm` does generate things in `config/` such as the `manifests/` directory, along with other patches. I'm not sure if these are also considered side effects that we don't want in the base `config/` directory.

Any artifacts that should be checked into source control that represent "a release" are totally cool to be "side effects" :) The only issue here was the modification of a different existing artifact that should *not* have been checked into source control since it wasn't actually part of the set of artifacts being produced for the OLM release.

What you've got here is perfect now. :)